{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16     16256935     .     G     A\n",
      "\n"
     ]
    },
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-35-c9260ea53f77>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0ms_line\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m     \u001b[0mqstr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'chr'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0ms_line\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m':'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0ms_line\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m':'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0ms_line\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0;34m':'\u001b[0m\u001b[0;34m+\u001b[0m\u001b[0ms_line\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m     \u001b[0mvariants\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mqstr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "\n",
    "f = open('dida.vcf', 'r')\n",
    "lines = f.readlines()\n",
    "\n",
    "variants = []\n",
    "for line in lines:\n",
    "    print(line)\n",
    "    s_line = line.strip().split('\\t')\n",
    "    qstr = 'chr'+s_line[0]+':'+s_line[1]+':'+s_line[2]+':'+s_line[3]\n",
    "    variants.append(qstr)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from varsome_api.client import VarSomeAPIClient\n",
    "nf = open('varsome_token', 'r')\n",
    "api.key= nf.readline().strip()\n",
    "api = VarSomeAPIClient(api_key)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2018-03-15 14:08:09,014 - varsome_api.client - INFO - Time between request and response 0:00:00.572425\n",
      "2018-03-15 14:08:09,016 - varsome_api.client - INFO - Content length 9312\n",
      "2018-03-15 14:08:09,396 - varsome_api.client - INFO - Time between request and response 0:00:00.956082\n",
      "2018-03-15 14:08:09,399 - varsome_api.client - INFO - Content length 38643\n",
      "2018-03-15 14:08:09,409 - varsome_api.client - INFO - Time between request and response 0:00:00.968202\n",
      "2018-03-15 14:08:09,411 - varsome_api.client - INFO - Content length 37992\n"
     ]
    }
   ],
   "source": [
    "results = api.batch_lookup(variants, params={'add-source-databases': 'dann-snvs, ncbi-dbsnp, dbnsfp-dbscsnv'}, ref_genome='hg19')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'CC', did you mean ##chr11:66299163:GA:C##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'T', did you mean ##chr12:76741975:A:C##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'AGGTA', did you mean ##chr12:76740163:AGTTA:A##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"The reference genome at the query position does not match the sequence given 'T', it matches the alt sequence 'C'\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"The reference genome at the query position does not match the sequence given 'C', it matches the alt sequence 'G'\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'C', did you mean ##chr16:56553655:T:G##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'AAA', did you mean ##chr15:49048432:TTT:AT##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n",
      "96\n",
      "97\n",
      "98\n",
      "99\n",
      "100\n",
      "101\n",
      "102\n",
      "103\n",
      "104\n",
      "105\n",
      "106\n",
      "107\n",
      "108\n",
      "109\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "116\n",
      "117\n",
      "118\n",
      "119\n",
      "120\n",
      "121\n",
      "122\n",
      "123\n",
      "124\n",
      "125\n",
      "126\n",
      "127\n",
      "128\n",
      "129\n",
      "130\n",
      "131\n",
      "132\n",
      "133\n",
      "134\n",
      "135\n",
      "136\n",
      "137\n",
      "138\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'GT', did you mean ##chrX:68836405:GG:G##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "143\n",
      "144\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'C', did you mean ##chr6:6449797:G:T##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "145\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"The reference genome at the query position does not match the sequence given 'C', it matches the alt sequence 'T'\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "146\n",
      "147\n",
      "148\n",
      "149\n",
      "150\n",
      "151\n",
      "152\n",
      "153\n",
      "154\n",
      "155\n",
      "156\n",
      "157\n",
      "158\n",
      "159\n",
      "160\n",
      "161\n",
      "162\n",
      "163\n",
      "164\n",
      "165\n",
      "166\n",
      "167\n",
      "168\n",
      "169\n",
      "170\n",
      "171\n",
      "172\n",
      "173\n",
      "174\n",
      "175\n",
      "176\n",
      "177\n",
      "178\n",
      "179\n",
      "180\n",
      "181\n",
      "182\n",
      "183\n",
      "184\n",
      "185\n",
      "186\n",
      "187\n",
      "188\n",
      "189\n",
      "190\n",
      "191\n",
      "192\n",
      "193\n",
      "194\n",
      "195\n",
      "196\n",
      "197\n",
      "198\n",
      "199\n",
      "200\n",
      "201\n",
      "202\n",
      "203\n",
      "204\n",
      "205\n",
      "206\n",
      "207\n",
      "208\n",
      "209\n",
      "210\n",
      "211\n",
      "212\n",
      "213\n",
      "214\n",
      "215\n",
      "216\n",
      "217\n",
      "218\n",
      "219\n",
      "220\n",
      "221\n",
      "222\n",
      "223\n",
      "224\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'G', did you mean ##chr7:150644884:C:GG##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "225\n",
      "226\n",
      "227\n",
      "228\n",
      "229\n",
      "230\n",
      "231\n",
      "232\n",
      "233\n",
      "234\n",
      "235\n",
      "236\n",
      "237\n",
      "238\n",
      "239\n",
      "240\n",
      "241\n",
      "242\n",
      "243\n",
      "244\n",
      "245\n",
      "246\n",
      "247\n",
      "248\n",
      "249\n",
      "250\n",
      "251\n",
      "252\n",
      "253\n",
      "254\n",
      "255\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'C', did you mean ##chr9:39995540:N:T##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "256\n",
      "257\n",
      "258\n",
      "259\n",
      "260\n",
      "261\n",
      "262\n",
      "263\n",
      "264\n",
      "265\n",
      "266\n",
      "267\n",
      "268\n",
      "269\n",
      "270\n",
      "271\n",
      "272\n",
      "273\n",
      "274\n",
      "275\n",
      "276\n",
      "277\n",
      "278\n",
      "279\n",
      "280\n",
      "281\n",
      "282\n",
      "283\n",
      "284\n",
      "285\n",
      "286\n",
      "287\n",
      "288\n",
      "289\n",
      "290\n",
      "291\n",
      "292\n",
      "293\n",
      "294\n",
      "295\n",
      "296\n",
      "297\n",
      "298\n",
      "299\n",
      "300\n",
      "301\n",
      "302\n",
      "303\n",
      "304\n",
      "305\n",
      "306\n",
      "307\n",
      "308\n",
      "309\n",
      "310\n",
      "311\n",
      "312\n",
      "313\n",
      "314\n",
      "315\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'GTTG', did you mean ##chr10:55581882:CGTT:G##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "316\n",
      "317\n",
      "318\n",
      "319\n",
      "320\n",
      "321\n",
      "322\n",
      "323\n",
      "324\n",
      "325\n",
      "326\n",
      "327\n",
      "328\n",
      "329\n",
      "330\n",
      "331\n",
      "332\n",
      "333\n",
      "334\n",
      "335\n",
      "336\n",
      "337\n",
      "338\n",
      "339\n",
      "340\n",
      "341\n",
      "342\n",
      "343\n",
      "344\n",
      "345\n",
      "346\n",
      "347\n",
      "348\n",
      "349\n",
      "350\n",
      "351\n",
      "352\n",
      "353\n",
      "354\n",
      "355\n",
      "356\n",
      "357\n",
      "358\n",
      "359\n",
      "360\n",
      "361\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'G', did you mean ##chr11:62380922:T:GG##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "362\n",
      "363\n",
      "364\n",
      "365\n",
      "366\n",
      "367\n",
      "368\n",
      "369\n",
      "370\n",
      "371\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "'chromosome'\n",
      "{'error': \"the reference genome at the query position does not match the sequence given 'GGAG', did you mean ##chr3:38593010:GAGA:G##?\"}\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "372\n",
      "373\n",
      "374\n",
      "375\n",
      "376\n",
      "377\n",
      "378\n",
      "379\n",
      "380\n",
      "381\n",
      "382\n",
      "383\n",
      "384\n",
      "385\n",
      "386\n",
      "387\n",
      "388\n",
      "389\n",
      "390\n",
      "391\n",
      "392\n",
      "393\n",
      "394\n",
      "395\n",
      "396\n",
      "397\n",
      "398\n",
      "399\n",
      "400\n",
      "401\n",
      "402\n",
      "403\n",
      "404\n",
      "405\n",
      "406\n",
      "407\n",
      "408\n",
      "409\n",
      "410\n",
      "411\n",
      "412\n",
      "413\n",
      "414\n",
      "415\n",
      "416\n",
      "417\n",
      "418\n",
      "419\n",
      "420\n",
      "421\n",
      "422\n",
      "423\n",
      "424\n",
      "425\n",
      "426\n",
      "427\n",
      "428\n",
      "429\n",
      "430\n",
      "431\n",
      "432\n",
      "433\n",
      "434\n",
      "435\n",
      "436\n",
      "437\n",
      "438\n",
      "439\n",
      "440\n",
      "441\n",
      "442\n",
      "443\n",
      "444\n",
      "445\n",
      "446\n",
      "447\n"
     ]
    }
   ],
   "source": [
    "fo = open('dida_dann.tsv', 'w')\n",
    "\n",
    "\n",
    "\n",
    "for i in range(len(lines)): \n",
    "    line = lines[i]\n",
    "    e = results[i]\n",
    "    print(i)\n",
    "    try:\n",
    "\n",
    "        fo.write(line.strip())\n",
    "        fo.write('\\t')\n",
    "        if 'chromosome' in e:   \n",
    "            if 'dann_snvs' in e:\n",
    "                fo.write(str(e['dann_snvs'][0]['dann_score']))\n",
    "                fo.write('\\n')\n",
    "            else:\n",
    "                fo.write('NA')\n",
    "                fo.write('\\n')\n",
    "        else:\n",
    "            fo.write('NA')\n",
    "            fo.write('\\n')\n",
    "            e['chromosome'][3:]\n",
    "                \n",
    "    except Exception as ex:\n",
    "        print('\\n')\n",
    "        print('\\n')\n",
    "        print(ex)\n",
    "        print(e)\n",
    "        print('\\n')\n",
    "        print('\\n')        "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
